### PR TITLE
Feat/schedule remainder mail to assignee for induction

### PR DESCRIPTION
**Pull request by nishant22029**
Added a cron job to send reminder emails to assignees. The cron job is scheduled to run daily at 8:00 AM.

### CODE REVIEW

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+  try {
+
+    $currentDate = date('Y-m-d');
+
+    $activity_assignees = \Civi\Api4\Activity::get(TRUE)
+        ->addSelect('target_contact_id', 'activity_contact.contact_id', 'activity_date_time')
+        ->addJoin('ActivityContact AS activity_contact', 'LEFT', ['activity_contact.record_type_id', '=', 1])
+        ->addWhere('activity_type_id', '=', 57)
+        ->addWhere('status_id', '=', 1)
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:36]: can we create a variable for this like we have done for the other two above?

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        ->addWhere('activity_date_time', '>=', $currentDate . ' 00:00:00')
+        ->addWhere('activity_date_time', '<=', $currentDate . ' 23:59:59')
+        ->setLimit(25)
+        ->execute();
+
+
+    foreach ($activity_assignees as $assignee) {
+        $assignee_contact_id = $assignee['activity_contact.contact_id'];
+        $volunteer_contect_id = $assignee['target_contact_id'][0];
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:45]:
```suggestion
        $volunteer_contact_id = $assignee['target_contact_id'][0];
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            'from' => $from,
+            'toEmail' => $assignee_details[0]['email.email'],
+            'toName' => $assignee_details[0]['display_name'],
+            'replyTo' => $from,
+            'html' => $html,
+            // 'messageTemplateID' => 76,
+          ];
+        try{
+            $result = CRM_Utils_Mail::send($mailParams);
+            error_log('$result' . print_r($result, true));
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:84]: remove

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        }catch (CiviCRM_API3_Exception $e) {
+            error_log('Goonj Cron Job: API error - ' . $e->getMessage());
+        }
+    }
+    error_log('Goonj Cron Job: Job completed successfully');
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:89]: remove

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+} catch (CiviCRM_API3_Exception $e) {
+    error_log('Goonj Cron Job: API error - ' . $e->getMessage());
+}
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:92]: check indentation of this function

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    $currentDate = date('Y-m-d');
+
+    $activity_assignees = \Civi\Api4\Activity::get(TRUE)
+        ->addSelect('target_contact_id', 'activity_contact.contact_id', 'activity_date_time')
+        ->addJoin('ActivityContact AS activity_contact', 'LEFT', ['activity_contact.record_type_id', '=', 1])
+        ->addWhere('activity_type_id', '=', 57)
+        ->addWhere('status_id', '=', 1)
+        ->addWhere('activity_date_time', '>=', $currentDate . ' 00:00:00')
+        ->addWhere('activity_date_time', '<=', $currentDate . ' 23:59:59')
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:38]: Can we define two variables like this:
```php
// Start of the day
$startOfDay = new DateTime('today midnight');

// End of the day
$endOfDay = new DateTime('tomorrow midnight -1 second');
```

and then do:

```suggestion
        ->addWhere('activity_date_time', '>=', $startOfDay->format('Y-m-d H:i:s'))
        ->addWhere('activity_date_time', '<=', $endOfDay->format('Y-m-d H:i:s'))
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  $returnValues = [];
+
+  try {
+
+    $currentDate = date('Y-m-d');
+
+    $activity_assignees = \Civi\Api4\Activity::get(TRUE)
+        ->addSelect('target_contact_id', 'activity_contact.contact_id', 'activity_date_time')
+        ->addJoin('ActivityContact AS activity_contact', 'LEFT', ['activity_contact.record_type_id', '=', 1])
+        ->addWhere('activity_type_id', '=', 57)
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:35]: sure

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+function civicrm_api3_goonjcustom_cron($params) {
+  $returnValues = [];
+
+  try {
+
+    $currentDate = date('Y-m-d');
+
+    $activity_assignees = \Civi\Api4\Activity::get(TRUE)
+        ->addSelect('target_contact_id', 'activity_contact.contact_id', 'activity_date_time')
+        ->addJoin('ActivityContact AS activity_contact', 'LEFT', ['activity_contact.record_type_id', '=', 1])
```
[the remark this answers is not in the document]
Comment (nishant22029) [reply wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:34]: sure

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php (hunk 6 above)
Comment (nishant22029) [reply wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/Cron.php:38]: sure updated

**Review by pokhiii — APPROVED** [review]

**Pull request merged**